src/Update/PowerToys.Update.cpp

```diff
@@ -169,13 +169,33 @@ bool InstallNewVersionStage2(std::wstring installer_path, std::wstring_view inst
         }
     }
 
+    for (const auto& entry : fs::directory_iterator(updating::get_pending_updates_path()))
+    {
+        auto entryPath = entry.path().wstring();
+        std::transform(entryPath.begin(), entryPath.end(), entryPath.begin(), ::towlower);
+
+        if ((entryPath.ends_with(L".msi") || entryPath.ends_with(L".exe")) && installer_path.find(entryPath) == std::string::npos)
+        {
+            std::error_code err;
+            fs::remove(entry, err);
+            if (err.value())
+            {
+                Logger::warn("Failed to delete file {}. {}", entry.path().string(), err.message());
+            }
+        }
+    }
+
     if (!success)
     {
         return false;
     }
 
-    std::error_code _;
-    fs::remove(installer_path, _);
+    std::error_code err;
+    fs::remove(installer_path, err);
```

Comment:
Why don't we delete the installer of the current update in line 180, if it was successful? This code part here seems to be obsolete and duplicated. (And if we don't delete it here it can be deleted by the other code part before on the next update.)

I think the code parts can be merged.

Reply:
I agree: this is done mostly for readability and to avoid complex or nested if.
What do you think @yuyoyuppe ?

Reply:
yep, that sounds logical, just need to make sure that it was successful.